cmd/controlplane/api.go

```diff
@@ -135,48 +153,139 @@ func newClientForAPI(ctx context.Context, r *rest.Config, scheme *runtime.Scheme
 	}
 
+	go func() {
+		if err := mgr.Start(ctx); err != nil {
+			panic(fmt.Errorf("start Argo Rollouts AnalyisRun API manager: %w", err))
+		}
+	}()
+
+	return restCfg, mgr.GetClient(), nil
+}
+
+func registerKargoIndexers(ctx context.Context, mgr ctrl.Manager) error {
 	// Index Promotions by Stage
 	if err := kubeclient.IndexPromotionsByStage(ctx, mgr); err != nil {
-		return nil, fmt.Errorf("index Promotions by Stage: %w", err)
+		return fmt.Errorf("index Promotions by Stage: %w", err)
 	}
 
 	// Index Freight by Warehouse
 	if err := kubeclient.IndexFreightByWarehouse(ctx, mgr); err != nil {
-		return nil, fmt.Errorf("index Freight by Warehouse: %w", err)
+		return fmt.Errorf("index Freight by Warehouse: %w", err)
 	}
 
 	// Index Freight by Stages in which it has been verified
 	if err := kubeclient.IndexFreightByVerifiedStages(ctx, mgr); err != nil {
-		return nil, fmt.Errorf("index Freight by Stages in which it has been verified: %w", err)
+		return fmt.Errorf("index Freight by Stages in which it has been verified: %w", err)
 	}
 
 	// Index Freight by Stages for which it is approved
-	if err :=
-		kubeclient.IndexFreightByApprovedStages(ctx, mgr); err != nil {
-		return nil, fmt.Errorf("index Freight by Stages for which it has been approved: %w", err)
+	if err := kubeclient.IndexFreightByApprovedStages(ctx, mgr); err != nil {
+		return fmt.Errorf("index Freight by Stages for which it has been approved: %w", err)
 	}
 
 	// Index ServiceAccounts by ODIC email
 	if err := kubeclient.IndexServiceAccountsByOIDCEmail(ctx, mgr); err != nil {
-		return nil, fmt.Errorf("index ServiceAccounts by OIDC email: %w", err)
+		return fmt.Errorf("index ServiceAccounts by OIDC email: %w", err)
 	}
+
 	// Index ServiceAccounts by OIDC groups
 	if err := kubeclient.IndexServiceAccountsByOIDCGroups(ctx, mgr); err != nil {
-		return nil, fmt.Errorf("index ServiceAccounts by OIDC groups: %w", err)
+		return fmt.Errorf("index ServiceAccounts by OIDC groups: %w", err)
 	}
+
 	// Index ServiceAccounts by OIDC subjects
 	if err := kubeclient.IndexServiceAccountsByOIDCSubjects(ctx, mgr); err != nil {
-		return nil, fmt.Errorf("index ServiceAccounts by OIDC subjects: %w", err)
+		return fmt.Errorf("index ServiceAccounts by OIDC subjects: %w", err)
 	}
 
-	go func() {
-		if err := mgr.Start(ctx); err != nil {
-			panic(fmt.Errorf("start manager: %w", err))
-		}
-	}()
+	return nil
+}
 
-	return mgr.GetClient(), nil
+func newWrappedKubernetesClient(
+	ctx context.Context,
+	restCfg *rest.Config,
+	internalClient client.Client,
+	serverCfg config.ServerConfig,
+) (kubernetes.Client, error) {
+	kubeClientOptions := kubernetes.ClientOptions{
+		NewInternalClient: func(context.Context, *rest.Config, *runtime.Scheme) (client.Client, error) {
+			return internalClient, nil
+		},
+	}
+	if serverCfg.OIDCConfig != nil {
+		kubeClientOptions.GlobalServiceAccountNamespaces = serverCfg.OIDCConfig.GlobalServiceAccountNamespaces
+	}
```

Comment:
@krancour I could not figure out if this is applicable to the Argo Rollouts specific client.

Reply:
Oh wow. Good question. This throws a monkey wrench into things... 🤦‍♂️ 

Question is... how do can we ask this wrapper to perform a subject access review when the SAs live in one cluster (Kargo control plane) and the Rollouts control plane lives in another cluster...

I think the answer is, we don't. It would be too complicated and too dangerous to have RBAC in one cluster dictating permissions in another cluster.

I was already fairly certain that we'd eventually kill off the separate Rollouts client, and I think this is the impetus that seals the deal.

I say rip out the extra client... back this branch up just in case.

@gdsoumya just want to check that you agree with me here. This is why I had pinged you on Slack... in the controller, we left a separate client for Rollouts, acknowledging we probably wouldn't need it. Here's @hiddeco has been _adding_ a client for Rollouts to the API server, but that's starting to make less and less sense and present new problems. Would you agree we should just not do this?

Reply:
Ripped it out again in 8c055fa69ff05f0e8a91497216b10cb493663685